### PR TITLE
fix(reset_password): Update formPrefill only if user enters an email address

### LINKED ISSUE #5293: signin page email is cleared after visiting /reset_password
STR

1. Go to /signin, fill out an email address and password
2. Click "Forgot password?"
3. Click "Remember password? Sign in."
4. On the /signin page, the email is cleared, the password is filled in.

forgotten-email

### PR DESCRIPTION
**Pull request by shane-tomlinson**
fixes #5293

### CODE REVIEW

**Comment by shane-tomlinson** [comment]
@mozilla/fxa-devs - r? This is PR is rebased off of #5294, I opened the other PR because the test changes to update reset_password to use selectors.js was somewhat bulky on its own.

File: tests/functional/reset_password.js [hunk 1]
```diff
@@ -128,18 +128,38 @@ define([
     },
 
     'open /reset_password page from /signin': function () {
+      const updatedEmail = 'testuser@testuser.com';
```
Comment (vladikoff) [inline tests/functional/reset_password.js]: nit: `testuser.com` a bit worried about that if this test evolves we might start sending traffic to that domain/email

File: app/scripts/views/reset_password.js [hunk 2]
```diff
@@ -39,12 +48,26 @@ define(function (require, exports, module) {
...
+      const email = this.getElementValue('.email');
+      // The email field is not pre-filled for the reset_password page,
+      // but if the user enters an email address, the entered email
+      // address should be propagated back to the signin page. If
+      // the user enters no email and instead clicks "Remember password?"
+      // immediately, the /signin page should have the original email.
+      // See https://github.com/mozilla/fxa/tree/master/features/FxA-72-reset-password
+      // and #5293
+      if (email) {
+        this._formPrefill.set({ email });
```
Comment (vladikoff) [inline app/scripts/views/reset_password.js]: 👍

File: tests/functional/reset_password.js (hunk 1 above)
Comment (shane-tomlinson) [inline tests/functional/reset_password.js]: I'll fix that.

**Pull request merged**